internal/controllers/clusterextension_controller.go

```diff
@@ -466,32 +479,36 @@ type DefaultInstalledBundleGetter struct {
 	helmclient.ActionClientGetter
 }
 
-func (d *DefaultInstalledBundleGetter) GetInstalledBundle(ctx context.Context, ext *ocv1alpha1.ClusterExtension) (*ocv1alpha1.BundleMetadata, error) {
+func (d *DefaultInstalledBundleGetter) GetInstalledBundle(ctx context.Context, ext *ocv1alpha1.ClusterExtension) (*ocv1alpha1.BundleMetadata, release.Status, error) {
+	status := release.StatusUnknown
 	cl, err := d.ActionClientFor(ctx, ext)
 	if err != nil {
-		return nil, err
+		return nil, status, err
 	}
 
-	rel, err := cl.Get(ext.GetName())
+	relhis, err := cl.History(ext.GetName())
```

Comment:
Refactor suggestion:
```go
relhis, err := cl.History(ext.GetName())
if errors.Is(err, driver.ErrReleaseNotFound)
	return nil, nil
}
if err != nil {
	return nil, err
}
for _, rel := range relhis {
	if rel.Info.Status == release.StatusDeployed {
 		return &ocv1alpha1.BundleMetadata{
 			Name:    rel.Labels[labels.BundleNameKey],
 			Version: rel.Labels[labels.BundleVersionKey],
	 	}, nil
 	}
 }
return nil, nil
```

Reply:
I was thinking that... but wondering if it makes it harder to read...

Reply:
IMO, its easier with my suggestion, but I realize this is subjective. 

- The first two `if` conditionals seem more straightforward to grok.
- The `return` inside the loop is _maybe_ slightly more difficult to read, but it eliminates the extra variable, `break` and assignment that would be required to return after the loop completes.